Kentor.AuthServices/Configuration/SPOptions.cs

```diff
@@ -217,9 +220,107 @@ public IdentityConfiguration SystemIdentityModelIdentityConfiguration
             }
         }
 
+        readonly Collection<ServiceCertificate> serviceCertificates = new Collection<ServiceCertificate>();
+
+        /// <summary>
+        /// Certificates used by the service provider for signing or decryption.
+        /// </summary>
+        public Collection<ServiceCertificate> ServiceCertificates
+        {
+            get
+            {
+                return serviceCertificates;
+            }
+        }
+
+        /// <summary>
+        /// Certificates valid for use in decryption
+        /// </summary>
+        public ReadOnlyCollection<X509Certificate2> DecryptionServiceCertificates
+        {
+            get
+            {
+                var decryptionCertificates = ServiceCertificates
+                    .Where(c => c.Use == CertificateUse.Encryption || c.Use == CertificateUse.Both)
+                    .Select(c => c.Certificate);
+
+                return decryptionCertificates.ToList().AsReadOnly();
+            }
+        }
+
+        /// <summary>
+        /// Certificate for use in signing outbound requests
+        /// </summary>
+        public X509Certificate2 SigningServiceCertificate
+        {
+            get
+            {
+                var decryptionCertificates = ServiceCertificates
```

Comment:
I'm looking at the `decryptionCertificates` variable - looks like code is copy pasted from property above. The conditions are changed, but not the var name.